src/commands/importFromRegistry.ts

```diff
@@ -0,0 +1,167 @@
+import * as vscode from "vscode";
+import { Keychain } from "../keychain";
+
+export async function importFromRegistry(scope?: vscode.ConfigurationScope) {
+  const cmd = require("node-cmd");
+  const vsWinReg = require("vscode-windows-registry");
+  const hkeyLocalMachine = "HKEY_LOCAL_MACHINE";
+
+  const config = vscode.workspace.getConfiguration("intersystems", scope);
+  const serverDefinitions: any = config.get("servers");
+
+  const newServerNames = new Array<string>();
+  const serversMissingUsernames = new Array<string>();
+  const cancellationToken = new vscode.CancellationTokenSource().token;
+  cancellationToken.onCancellationRequested(() => {
+    vscode.window.showInformationMessage("Cancelled server import.");
+  });
+
+  await cmd.get(
+    "reg query " +
+      hkeyLocalMachine +
+      "\\SOFTWARE\\Intersystems\\Cache\\Servers",
+    async (err, data, stderr) => {
+      // Prompt user for username (if not in registry) and password
+      data.split("\n").forEach((serverName) => {
+        // remove HKEY_LOCAL_MACHINE\ and whitespace from the server name
+        const path = serverName.substring(hkeyLocalMachine.length + 1).trim();
+
+        const originalName: string = serverName.split("\\").pop().trim();
+        // Enforce the rules from package.json on the server name
+        const name = originalName.toLowerCase().replace(/[^A-Za-z1-9-]/g, "~");
```

Comment:
Server Manager allows `a-z0-9-._~` so I'm not clear why this line apparently transforms legal characters 0, . and _ into ~